src/lib/support/ScopedBuffer.h

```diff
@@ -84,30 +89,40 @@ class ScopedMemoryBufferBase
     const void * Ptr() const { return mBuffer; }
 
     /**
-     * Releases the undelying buffer. Buffer stops being managed and will not be
+     * Releases the underlying buffer. Buffer stops being managed and will not be
      * auto-freed.
      */
     void * Release()
     {
-        void * buffer = mBuffer;
-        mBuffer       = nullptr;
+        void * buffer  = mBuffer;
+        mBuffer        = nullptr;
+        mAllocatedSize = 0;
         return buffer;
     }
 
     void Alloc(size_t size)
     {
         Free();
         mBuffer = Impl::MemoryAlloc(size);
+        if (mBuffer != nullptr)
+        {
+            mAllocatedSize = size;
+        }
     }
 
     void Calloc(size_t elementCount, size_t elementSize)
     {
         Free();
         mBuffer = Impl::MemoryCalloc(elementCount, elementSize);
+        if (mBuffer != nullptr)
+        {
+            mAllocatedSize = elementCount * elementSize;
+        }
     }
 
 private:
-    void * mBuffer = nullptr;
+    void * mBuffer        = nullptr;
+    size_t mAllocatedSize = 0;
```

Comment:
This seems to be only used in tests. Could we avoid having it? Since this is scopde, any code using  it presumably already has a 'size_t amount' that it allocates, so unless you pass it around you should not need the extra `size'

I would prefer to not allocate another integer just because tests require them (minor space saving)

Reply:
Will refactor